tracing-core/src/dispatcher.rs

```diff
@@ -434,6 +434,23 @@ impl Dispatch {
         self.subscriber.register_callsite(metadata)
     }
 
+    /// Returns the highest [verbosity level][level] that this [`Subscriber`] will
+    /// enable, or `None`, if the subscriber does not implement level-based
+    /// filtering or chooses not to implement this method.
+    ///
+    /// This calls the [`max_level_hint`] function on the [`Subscriber`]
+    /// that this `Dispatch` forwards to.
+    ///
+    ///
+    /// [level]: ../struct.Level.html
+    /// [`Subscriber`]: ../subscriber/trait.Subscriber.html
+    /// [`register_callsite`]: ../subscriber/trait.Subscriber.html#method.max_level_hint
+    // TODO(eliza): consider making this a public API?
```

Comment:
@LucioFranco @davidbarsky do we want to commit to making `Dispatch::max_level_hint` a public API? 

Do we see any need for user code to call the `max_level_hint` method on a particular `Dispatch`? This seems like a potential footgun, since it would only expose _that_ subscriber's max level, rather than the globally computed max level for all active subscribers — if there are other subscribers in the program, that dispatcher's hint might be lower than the max level that should be enabled globally.

Reply:
> Do we see any need for user code to call the max_level_hint method on a particular Dispatch?

I can't think of any right now.

> This seems like a potential footgun, since it would only expose that subscriber's max level, rather than the globally computed max level for all active subscribers — if there are other subscribers in the program, that dispatcher's hint might be lower than the max level that should be enabled globally.

Agreed. Let's be conservative in what is actually exposed for now. This can always be made public, it'd be _much_ harder to remove it.